eng/common/scripts/Helpers/PSModule-Helpers.ps1

```diff
@@ -55,68 +55,74 @@ function Install-ModuleIfNotInstalled()
   param(
     [string]$moduleName,
     [string]$version,
-    [string]$repositoryUrl = $DefaultPSRepositoryUrl
+    [string]$repositoryUrl
   )
 
-  # Check installed modules
-  $modules = (Get-Module -ListAvailable $moduleName)
-  if ($version -as [Version]) {
-    $modules = $modules.Where({ [Version]$_.Version -ge [Version]$version })
+  # List of modules+versions we want to replace with internal feed sources for reliability, security, etc.
+  $mirrorFeedOverrides = @{
+    'powershell-yaml;0.4.7' = 'https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-tools/nuget/v2'
```

Comment:
Do we care about the version?

Also is there any drawback to query this feed for everything first and then fallback? Do they allow multiple feeds?

Reply:
We only care because due to security constraints we can't set the feed as upstream in a way where we can make all the packages available. So unless I mirror every version, this simply reflects which package versions I have manually uploaded to our feed.

I like the idea of querying first and then adding fallbacks though, i.e. if one fails we go to the other. I can go and implement that. And then we wouldn't need to specify versions either.